datastore/postgres/updatevulnerabilities.go

```diff
@@ -117,6 +156,64 @@ func (s *MatcherStore) UpdateVulnerabilities(ctx context.Context, updater string
 		Str("ref", ref.String()).
 		Msg("update_operation created")
 
+	if delta {
+		// Get existing vulns
+		// The reason this still works even though the new update_operation
+		// is already created is because the latest_update_operation view isn't updated until
+		// the end of this function.
+		start = time.Now()
+		rows, err := s.pool.Query(ctx, selectExisting, updater)
+		if err != nil {
+			return uuid.Nil, fmt.Errorf("failed to get existing vulns: %w", err)
+		}
+		updateVulnerabilitiesCounter.WithLabelValues("selectExisting").Add(1)
+		updateVulnerabilitiesDuration.WithLabelValues("selectExisting").Observe(time.Since(start).Seconds())
+
+		oldVulns := make(map[string][]string)
+		for rows.Next() {
+			var tmpID int64
+			var ID, name string
+			err := rows.Scan(
+				&name,
+				&tmpID,
+			)
+
+			ID = strconv.FormatInt(tmpID, 10)
+			if err != nil {
+				return uuid.Nil, fmt.Errorf("failed to scan vulnerability: %w", err)
+			}
+			oldVulns[name] = append(oldVulns[name], ID)
+		}
+
+		if len(oldVulns) > 0 {
+			for _, v := range vulns {
+				// If we have an existing vuln in the new batch
+				// delete it from the oldVulns map so it doesn't
+				// get associated with the new update_operation.
+				delete(oldVulns, v.Name)
+			}
+			for _, delName := range deletedVulns {
+				// If we have an existing vuln that has been signaled
+				// as deleted by the updater then delete it so it doesn't
+				// get associated with the new update_operation.
+				delete(oldVulns, delName)
+			}
+		}
+		start = time.Now()
+		// Update oldVulns' update_operations
+		for _, vs := range oldVulns {
+			for _, vID := range vs {
+				_, err := tx.Exec(ctx, assocExisting, uoID, vID)
+				if err != nil {
+					return uuid.Nil, fmt.Errorf("could not update old vulnerability with new UO: %w", err)
+				}
+			}
+		}
```

Comment:
Two loops here feels kind of bad, but I guess it's okay. The alternative I hit on was building an indexed slice data structure, which is probably not worth it in a first implementation.